custom/ewsghana/reports/__init__.py

```diff
@@ -165,15 +163,50 @@ class MultiReport(DatespanMixin, CustomProjectReport, ProjectReportParametersMix
 
     @property
     @memoized
-    def location(self):
+    def root_location(self):
         from custom.ewsghana import ROOT_SITE_CODE
+        return get_object_or_404(
+            SQLLocation, site_code=ROOT_SITE_CODE, domain=self.domain, is_archived=False
+        )
+
+    @property
+    @memoized
+    def user_location(self):
+        user = self.request.couch_user
+        dm = user.get_domain_membership(self.domain)
+        if not dm:
+            return
+
+        if dm.location_id:
+            return get_object_or_404(
+                SQLLocation, domain=self.domain, location_id=dm.location_id, is_archived=False
+            )
+
+        try:
+            ews_extension = EWSExtension.objects.get(user_id=user.get_id, domain=self.domain)
```

Comment:
it would be better if this was the only line in the try/except since the rest doesn't raise a DoesNotExist exception